app/services/bulk_ingest_service/bulk_file_path_converter.rb

```diff
@@ -30,24 +30,53 @@ def ingestable_files
 
     def ingestable_files_with_captions
       ingestable_files.map do |f|
-        next unless (vtt = matching_vtt(f))
-        vtt_path = Pathname(vtt)
-        f.container_attributes[:files] = [
-          IngestableFile.new(
-            file_path: vtt_path,
-            mime_type: "text/vtt",
-            original_filename: vtt_path.basename.to_s,
-            use: Valkyrie::Vocab::PCDMUse.Caption
-          )
-        ]
+        vtts = matching_vtts(f)
+        next if vtts.blank?
+        vtt_paths = vtts.map { |vtt| Pathname(vtt) }
+        f.container_attributes[:files] = vtt_paths.map do |vtt_path|
+          build_ingestable_vtt(vtt_path)
+        end
         f
       end
     end
 
-    def matching_vtt(file)
-      caption_files.find do |cf|
+    def matching_vtts(file)
+      caption_files.select do |cf|
         cf.starts_with?(file.file_path.sub_ext("").to_s)
       end
     end
+
+    def build_ingestable_vtt(vtt_path)
+      change_set = ChangeSet.for(
+        FileMetadata.new,
+        change_set_param: "caption",
+        file: uploaded_file_struct(vtt_path)
+      )
+      change_set.validate(
+        caption_language: infer_language(vtt_path),
+        original_language_caption: infer_original_language(vtt_path)
+      )
+      change_set.to_ingestable_file
+    end
+
+    # mimic the UploadedFile expected by CaptionChangeSet
+    def uploaded_file_struct(vtt_path)
+      Struct.new(
+        :path, :content_type, :original_filename
```

Comment:
Clever! Can you use `IngestableFile.new(file_path: vtt_path, mime_type: "text/vtt", original_filename: vtt_path.basename.to_s)`? It's a little odd, but one of the use cases for IngestableFile was to handle cases where we don't have one of those Rails file uploads.

Reply:
Sure thing. It's just `IngestableFile` all the way down.